packages/bierzo-wallet/src/logic/account.ts

```diff
@@ -42,6 +46,19 @@ export async function lookupRecipientAddressByName(
   throw new Error("No BNS connection found");
 }
 
-export function isIov(username: string): boolean {
-  return username.endsWith("*iov");
+export function isValidIov(username: string): "valid" | "not_iov" | "too_short" | "too_long" | "wrong_chars" {
+  if (!isIov(username)) return "not_iov";
+
+  const usernameLength = username.length - "*iov".length;
```

Comment:
To make the logic easier, can you split by `*` first?

```ts
const parts = username.split("*");
if (parts.lenth !== 2) return "wrong_number_of_asterisks";
const [name, namespace] = parts;
```

This is also more future-proof since we will add different namespaces soon after launch.